File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/commands/CommandHandler.java
package me.lokka30.levelledmobs.bukkit.commands;

import java.util.HashSet;
import java.util.Set;
import me.lokka30.levelledmobs.bukkit.commands.levelledmobs.LevelledMobsBaseCommand;
import me.lokka30.levelledmobs.bukkit.utils.Log;
import org.jetbrains.annotations.NotNull;

public final class CommandHandler {

    /* vars */

    private final Set<BaseCommandWrapper> baseCommands = new HashSet<>();

    /* constructors */

    public CommandHandler() {
        baseCommands.addAll(Set.of(
            new LevelledMobsBaseCommand()
        ));
    }

    /* methods */

    public boolean load() {
        Log.inf("Registering commands.");
        for(var baseCommand : getBaseCommands()) {
            if(!baseCommand.register())
                return false;
        }
        return true;
    }

    /* getters and setters */

    @NotNull
    public Set<BaseCommandWrapper> getBaseCommands() {
        return baseCommands;
    }

}

File: levelledmobs-plugin-bukkit/src/main/kotlin/me/lokka30/levelledmobs/bukkit/commands/CommandHandler.kt
package me.lokka30.levelledmobs.bukkit.commands

import me.lokka30.levelledmobs.bukkit.LevelledMobs.Companion.logInf
import me.lokka30.levelledmobs.bukkit.commands.levelledmobs.LevelledMobsCommand

class CommandHandler {

    /*
    A set of base commands which are provided by LevelledMobs.
     */
    val baseCommands = mutableSetOf<BaseCommandWrapper>(LevelledMobsCommand())

    /*
    Register all base commands for LevelledMobs.

    Important note: Each command must be registered in plugin.yml, otherwise Bukkit will allow it to
    be registered.
     */
    fun load() {
        logInf("Registering commands...")
        baseCommands.forEach(BaseCommandWrapper::register)
    }

}